src/Controls/src/Core/RadioButtonGroupController.cs

```diff
@@ -27,21 +28,20 @@ public RadioButtonGroupController(Compatibility.Layout<View> layout)
 				UpdateGroupNames(layout, _groupName);
 			}
 
-			MessagingCenter.Subscribe<RadioButton, RadioButtonGroupSelectionChanged>(this,
-				RadioButtonGroup.GroupSelectionChangedMessage, HandleRadioButtonGroupSelectionChanged);
-			MessagingCenter.Subscribe<RadioButton, RadioButtonGroupNameChanged>(this, RadioButton.GroupNameChangedMessage,
-				HandleRadioButtonGroupNameChanged);
-			MessagingCenter.Subscribe<RadioButton, RadioButtonValueChanged>(this, RadioButton.ValueChangedMessage,
-				HandleRadioButtonValueChanged);
+			WeakReferenceMessenger.Default.Register<RadioButtonGroupController, RadioButtonGroupSelectionChanged>(this, HandleRadioButtonGroupSelectionChanged);
```

Comment:
These registrations are introducing closures, and also the `controller` parameter in the handlers doesn't actually seem to be used at all? Those could drop that parameter, and then the handlers here could be `(r, m) => r.Handle___(m)`, which would just cache a single delegate and avoid allocations.

Reply:
The controller parameter isn't being used _yet_; I was hoping to convert the HandleX methods to `static` later (rather than having them be instance methods on RBGController). 

I guess that's a question, though - which is better for performance? Instance methods (and don't pass in the `controller` parameter), or static methods? My assumption was that the static option was better.

Reply:
The most efficient solution is one that allows the registration to be allocation free, which needs to (1) not be creating a closure, so not capturing anything, and (2) allow Roslyn to cache and reuse the delegate instance. The way the compiler works right now, the method group syntax will always result in a new delegate instance being allocated. So the optimal way to register handlers (without using `IRecipient<TMessage>`), would be to use an instance method being invoked directly on the input recipient. This will alow the delegate to be stateless and reusable, and there will also be no allocations due to closures 😊

Reply:
Cool, I'll fix that (and change the RadioButton suff back to instance methods, too). 

> optimal way to register handlers (without using IRecipient<TMessage>)
I'd much rather use IRecipient for some of these (one RegisterAll call vs 4 for things like the AlertManager) - what are the performance implications if I go that direction?

Reply:
So, there are a few considerations with using `IRecipient<TMessage>`, I'll try to go through all of them:

- Once a handler is registered, they're mostly equivalent. Invoking a handler that was registered through `IRecipient<TMessage>` is actually ever so slightly faster, as there's a hot path for it in the broadcast code that allows the messenger to skip the callvirt to invoke the delegate. I wouldn't say it's a major difference though.
- If you register handlers individually (that is, using `IMessenger.Register<TMessage>(IRecipient<TMessage>)`), the registration is free and super fast, basically the same as what you're doing now manually.
- If you want to use the `RegisterAll` extension, that comes with some caveats. It either needs the source-generated registration methods (generated by the generator that ships with the MVVM Toolkit) to be available (meaning you either have to disable trimming, or manually add annotations to keep that generated method), or it will fallback to a path using a compiled LINQ Expressions, which will have a fair bit of overhead for the first invocation.
- Another aspect with `IRecipient<TMessage>` is that it adds a public interface to the recipient type. If that type is public, that's an obvious implementation detail leak.

Due to the last 3 points I would generally recommend sticking with manual registration in this scenario. Hope this helps, let me know if you have any other questions on this! 😄

Reply:
This is great, thanks for the thorough answer!
For our purposes, I think manual registration is the right answer.